bool farLeft
bool farRight
bool nearLeft
bool nearRight